internal/controller/clientpool/clientpool.go

```diff
@@ -25,17 +29,23 @@ type ClientPoolKey struct {
 	Namespace string
 }
 
+type ClientInfo struct {
+	Client  sdkclient.Client
+	TLS     *tls.Config // Storing the TLS config associated with the client to check certificate expiration. If the certificate is expired, a new client will be created.
+	PEMCert []byte      // Store the original PEM-encoded certificate for expiration checks
```

Comment:
Nit: even though this is already an internal package, it's good hygiene to avoid public symbols unless they need to be referenced by other packages. That way if a public package aliases this type, the fields do not accidentally become part of the public interface.

```suggestion
	client  sdkclient.Client
	tls     *tls.Config // Storing the TLS config associated with the client to check certificate expiration. If the certificate is expired, a new client will be created.
	pemCert []byte      // Store the original PEM-encoded certificate for expiration checks
```

Reply:
(same thing for `ClientPoolKey` fields, though I didn't notice those before)